src/Validator/Validator.php

```diff
@@ -59,8 +59,8 @@ public function isValidMatch(UrlMatch $match): bool
             return $this->isAllowedScheme($scheme);
         }
 
-        if (!$this->matchEmails && $this->isEmail($match)) {
-            return false;
+        if ($this->isEmail($match)) {
+            return $this->matchEmails;
```

Comment:
Could you please explain what issue does it solve?

For me it looks like it behave the same as previous condition. Apart of the fact that previous condition will work faster in cases when `$this->matchEmails = true` as it will not perform unnecessary `$this->isEmail($match)` call. Which is probably most of the library use cases.